TCMManagement/Controllers/AppointmentController.cs

```diff
@@ -1,38 +1,39 @@
-using System.Collections.Generic;
-using System.Linq;
-using System.Web.Http;
-using System.Net.Http;
-using TCMManagement.BusinessLayer;
-using TCMManagement.Models;
-using static TCMManagement.BusinessLayer.Constants;
-
-namespace TCMManagement.Controllers
-{
-    /// <summary>
-    /// All Controllers should follow CRUD sequence so it will be easier to maintain the code.
-    /// </summary>
-    public class AppointmentController : ApiController
-    {
-        private IEntityServices<Appointment> appointmentService;
-
-        public AppointmentController() {
-            appointmentService = new AppointmentService();
-        }
-
-        public AppointmentController(IEntityServices<Appointment> service)
-        {
-            appointmentService = service;
-        }
-
-        // Comment our for now, easier to test
-        // [Authorize(Roles = "Receptionist")]
-        [HttpPost]
-        public IHttpActionResult AddAppointment(Appointment a)
-        {
-            return Ok(appointmentService.CreateItem(a));
-        }
-
-        // querystring = "?person=1"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web.Http;
+using System.Net.Http;
+using TCMManagement.BusinessLayer;
+using TCMManagement.Models;
+using static TCMManagement.BusinessLayer.Constants;
+
+namespace TCMManagement.Controllers
+{
+    /// <summary>
+    /// All Controllers should follow CRUD sequence so it will be easier to maintain the code.
+    /// </summary>
+    public class AppointmentController : ApiController
+    {
+        private IEntityServices<Appointment> appointmentService;
+
+        public AppointmentController() {
+            appointmentService = new AppointmentService();
+        }
+
+        public AppointmentController(IEntityServices<Appointment> service)
+        {
+            appointmentService = service;
+        }
+
+        // Comment our for now, easier to test
+        // [Authorize(Roles = "Receptionist")]
+        [HttpPost]
+        public IHttpActionResult AddAppointment(Appointment a)
+        {
+            return Ok(appointmentService.CreateItem(a));

```

Comment:
Are we going to return OK(null) if CreateItem(a) return null?
Doesn't sound right to me.

Reply:
Change it to similar as personController::AddPerson

Reply:
Yeah, I think it will work.
I don't think we should return OK if it is null.
Right?
We can also throw back a conflict exception if you prefer.
-- I am not sure how to do our exception strategy yet.
-- I did a little bit in our PersonController just to experiment.